From 0c97e2ec2a9f11e4d2b130ac01928ee44b1cd4b3 Mon Sep 17 00:00:00 2001
From: Girija P <gpandiya@qti.qualcomm.com>
Date: Thu, 8 Jan 2026 00:11:48 +0530
Subject: [PATCH 15/54] Controller db duplication check modified to update new
 interface name instead of dropping cmdu Intel vendor specific TLV handling

Signed-off-by: Girija P <gpandiya@qti.qualcomm.com>
---
 controller/src/beerocks/master/controller.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/controller/src/beerocks/master/controller.cpp b/controller/src/beerocks/master/controller.cpp
index 56d393e52..13b329aca 100755
--- a/controller/src/beerocks/master/controller.cpp
+++ b/controller/src/beerocks/master/controller.cpp
@@ -3184,7 +3184,8 @@ bool Controller::handle_intel_slave_join(
             LOG(ERROR) << "Mac duplication detected between "
                        << database.get_radio_iface_name(radio_mac) << " and "
                        << notification->hostap().iface_name;
-            return false;
+	    LOG(ERROR) << "Update database with latest iface_name";
+	    database.set_radio_iface_name(radio_mac, notification->hostap().iface_name);
         }
     } else {
         database.add_radio(radio_mac, bridge_mac);
-- 
2.34.1

